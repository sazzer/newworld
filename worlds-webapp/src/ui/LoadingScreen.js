// @flow

import React from 'react';
import {Dimmer, Loader, Segment} from 'semantic-ui-react';
import {Interpolate} from 'react-i18next';

/**
 * Component to show a loading screen
 */
export default function LoadingScreen() {
    return (
        <Segment raised style={{minHeight: "10em"}}>
            <Dimmer active inverted>
                <Loader size='large' inline="centered">
                    <Interpolate i18nKey="loading.message" />
                </Loader>
            </Dimmer>
        </Segment>
    );
}
