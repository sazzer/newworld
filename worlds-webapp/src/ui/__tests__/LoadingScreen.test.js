// @flow

import React from 'react';
import {shallow} from 'enzyme';
import LoadingScreen from '../LoadingScreen';

/** Create the component to test */
function setup() {
    return {
        element: shallow(<LoadingScreen />)
    };
}

describe('LoadingScreen', () => {
    it('Renders correctly', () => {
        const { element } = setup();

        expect(element).toMatchSnapshot();
    });
});
