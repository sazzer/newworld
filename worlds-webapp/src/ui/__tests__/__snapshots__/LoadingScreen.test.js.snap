// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingScreen Renders correctly 1`] = `
<Segment
  raised={true}
  style={
    Object {
      "minHeight": "10em",
    }
  }
>
  <Dimmer
    active={true}
    inverted={true}
  >
    <Loader
      inline="centered"
      size="large"
    >
      <Interpolate
        className=""
        i18nKey="loading.message"
      />
    </Loader>
  </Dimmer>
</Segment>
`;
